next_generation/pint_server/app.py

```diff
@@ -323,13 +311,13 @@ def get_provider_images(provider):
 
 
 def get_data_version_for_provider_category(provider, category):
-    column_name = provider + category
-    versions = VersionsModel.query.all()[0]
-    try:
-        getattr(versions, column_name)
-    except AttributeError:
+    tablename = provider + category
+    version = VersionsModel.query.filter(
+        VersionsModel.tablename == tablename).first()
```

Comment:
Maybe consider using `one()` instead of first here which raises exceptions if there are None or more than one... Could allow you to simplify this routine to be just a return statement with exception handling that calls abort for the None, or multiples found cases... Just a thought ;-)

Reply:
The reason I used first() here is because I want to explicitly check for no result so we can return a 404 to match the current Pint Server behavior.

Reply:
That is the point of the exception handing of `one()` - https://docs.sqlalchemy.org/en/12/orm/query.html#sqlalchemy.orm.query.Query.one

It raises sqlalchemy.orm.exc.NoResultFound for no results found, and sqlalchemy.orm.exc.MultipleResultsFound for multiple results found, so you can handle both scenarios appropriately using exception handling without need to do an explicit check.

Reply:
Sound good. Let me give it a try.